examples/chef/chef.py

```diff
@@ -164,40 +236,190 @@ def main(argv: Sequence[str]) -> None:
                       action="store_true", dest="do_rpc_console")
     parser.add_option("-y", "--tty", help="Enumerated USB tty/serial interface enumerated for your physical device. E.g.: /dev/ACM0",
                       dest="tty", metavar="TTY", default=None)
+    parser.add_option("", "--generate_zzz", help="Populates zzz_generated/chef/<DEVICE_TYPE>/zap-generated with output of ZAP tool for every device in examples/chef/devices. If this flag is set, all other arguments are ignored except for --bootstrap_zap and --validate_zzz.",
+                      dest="generate_zzz", action="store_true")
+    parser.add_option("", "--validate_zzz", help="Checks if cached ZAP output needs to be regenrated, for use in CI. If this flag is set, all other arguments are ignored.",
+                      dest="validate_zzz", action="store_true")
+    parser.add_option("", "--use_zzz", help="Use pre generated output from the ZAP tool found in the zzz_generated folder. Used to decrease execution time of CI/CD jobs",
+                      dest="use_zzz", action="store_true")
+    parser.add_option("", "--build_all", help="For use in CD only. Builds and bundles all chef examples for the specified platform. Uses --use_zzz. Chef exits after completion.",
+                      dest="build_all", action="store_true")
+    parser.add_option(
+        "", "--ci", help="Builds Chef examples defined in _CI_ALLOW_LIST. Uses --use_zzz. Uses specified target from -t. Chef exits after completion.", dest="ci", action="store_true")
 
     options, _ = parser.parse_args(argv)
 
     splash()
 
+    #
+    # Validate zzz_generated
+    #
+
+    if options.validate_zzz:
+        fix_instructions = """Cached files out of date!
+        Please:
+        bootstrap, activate, and install zap
+        run chef with the flag --generate_zzz
+        git add examples/chef/zzz_generated
+        git add examples/chef/cimanifes.json
+        Ensure you are running with the latest version of ZAP from master!"""
+        ci_manifest = generate_device_manifest()
+        with open(_CI_MANIFEST_FILE_NAME, "r", encoding="utf-8") as ci_manifest_file:
+            cached_manifest = json.loads(ci_manifest_file.read())
+        for device in ci_manifest:
+            if device != "zap_commit":
+                try:
```

Comment:
This could be further flattened by removing `try` and testing with `in` . It should also be updated to verify there are no extra files in the manifest or output directory.